libraries/AP_Logger/LogFile.cpp

```diff
@@ -607,34 +607,57 @@ void AP_Logger::Write_Proximity(AP_Proximity &proximity)
         return;
     }
 
-    AP_Proximity::Proximity_Distance_Array dist_array {};
-    proximity.get_horizontal_distances(dist_array);
+    AP_Proximity::Proximity_Distance_Array dist_array {}, filt_dist_array {};
+    for (uint8_t i = 0; i < proximity.get_layers(); i++) {
+        const bool active = proximity.get_active_layer_distances(i, dist_array, filt_dist_array);
+        if (!active) {
+            // nothing on this layer
+            continue;
+        }
+        float dist_up;
+        if (!proximity.get_upward_distance(dist_up)) {
+            dist_up = 0.0f;
+        }
 
-    float dist_up;
-    if (!proximity.get_upward_distance(dist_up)) {
-        dist_up = 0.0f;
+        float close_ang = 0.0f, close_dist = 0.0f;
```

Comment:
```suggestion
        float closest_ang = 0.0f, closest_dist = 0.0f;
```

I think we can afford an extra two characters on these :-)